Userland/Libraries/LibThread/Thread.h

```diff
@@ -33,7 +33,7 @@ class Thread final : public Core::Object {
 
 private:
     explicit Thread(Function<int()> action, StringView thread_name = nullptr);
-    Function<int()> m_action;
+    Function<intptr_t()> m_action;
```

Comment:
It's not obvious to me why this is needed - in fact, the constructor just a line above this still takes a `Function<int()>` which is used as the value for `m_action` - so this doesn't look like "Fix int to pointer conversion"...? Please explain in the commit message what's going on here.

Reply:
Silly me forgot to update the constructor (and possibly all the callers). That return value is effectively passed over to `void pthread_exit(void *value_ptr)`, so that `int` needs to be an `intptr_t` to not risk losing bits for `pthread_join()`.